packages/mimir/src/rules/parameter-properties.ts

```diff
@@ -0,0 +1,294 @@
+                                construct,
+                                param,
+                                getLineBreakStyle(this.context.sourceFile),
+                            ),
+                        );
+                } else if (allPropsCanBeParamProps && !construct.parameters.every(isParameterProperty)) {
+                    for (const param of construct.parameters.filter((p) => !isParameterProperty(p)))
+                        this.addFailure(
+                            param.pos,
+                            param.end,
+                            FAILURE_STRINGS[Mode.Consistent].canBeParamPropsOnly,
+                            getFixerForLonghandProp(param, construct),
+                        );
+                }
+        }
+    }
+}
+
+function canBeParameterProperty(param: ts.ParameterDeclaration, construct: ts.ConstructorDeclaration): boolean {
+    if (!isIdentifier(param.name) || param.dotDotDotToken) return false;
+    const member = construct.parent!.members.find((mem) => isPropertyMatchForParam(mem, param));
+    if (member && !member.decorators) {
+        for (const stmt of statementsMinusDirectivesAndSuper(construct)) {
+            if (isStatementWithPossibleSideEffects(stmt, param)) return false;
+            if (isSimpleParamToPropAssignment(stmt, param)) return true;
+        }
+    }
+
+    return false;
+}
+
+function getFixerForDisallowedParameterProp(
+    construct: ts.ConstructorDeclaration,
+    param: ts.ParameterDeclaration,
+    lineBreak: string,
+): Replacement[] {
+    return [
+        /* Add assignment after directives and super() call (if they exists) */
+        Replacement.append(
+            getEndOfDirectivesAndSuper(construct),
+            `${lineBreak}this.${param.name.getText()} = ${param.name.getText()};`,
+        ),
+
+        /* Add property to class */
+        Replacement.append(construct.getStart(), getPropertyTextFromParam(param) + lineBreak),
+
+        /* Finally, delete modifiers from the parameter prop */
+        Replacement.delete(param.decorators ? param.modifiers![0].getStart() : param.pos, param.name.getStart()),
+    ];
+}
+
+function getFixerForLonghandProp(param: ts.ParameterDeclaration, construct: ts.ConstructorDeclaration): Replacement[] {
+    /* Class member and assignment to be removed */
+    const member = construct
+        .parent!.members.filter(isPropertyDeclaration)
+        .find((prop) => getPropertyName(prop.name) === param.name.getText())!;
+    const assignment = construct
+        .body!.statements.filter(isExpressionStatement)
+        .find((stmt) => isSimpleParamToPropAssignment(stmt, param))!;
+
+    return [
+        Replacement.delete(member.getStart(), member.end),
+        Replacement.delete(assignment.getStart(), assignment.end),
+
+        /* Prepend access modifiers to parameter declaration */
+        Replacement.append(
+            param.decorators ? param.decorators.end + 1 : param.getStart(),
```

Comment:
the `+1` here is not correct. In the following example it would insert the modifiers into the parameter name:
```ts
class Foo {
  constructor(@decorator()foo: string) {
  }
}
```